lib/util/props.js

```diff
@@ -5,6 +5,8 @@
 
 const astUtil = require('./ast');
 
+const PROP_TYPE_FUNCTIONS = ['object', 'shape', 'instanceOf', 'oneOf', 'oneOfType', 'objectOf', 'arrayOf', 'exact'];
```

Comment:
I'm not sure whether there's a better way of checking this, instead of hard-coding the list?

Reply:
something like
```js
import PropTypes from 'prop-types';
Object.keys(PropTypes)
```
?

Reply:
That would include prop-types which are not functions, e.g. `string`.

Seeing as these are unlikely to change much, I think I'm happy with this, if you are?

Reply:
True - but they do indeed change; I just added a new one a couple weeks ago (see #2188 where this technique actually prevented a bug)